buildSrc/src/main/groovy/com/uber/okbuck/core/model/android/AndroidTarget.groovy

```diff
@@ -177,11 +177,11 @@ abstract class AndroidTarget extends JavaLibTarget {
 
         Map<String, ClassField> extraBuildConfig = [:]
 
-        baseVariant.buildType.buildConfigFields.collect { String key, ClassField classField ->
+        baseVariant.mergedFlavor.buildConfigFields.collect { String key, ClassField classField ->
```

Comment:
My assumption seems incorrect. Gradle seems to give precedence to buildTypes. So 2 should be the correct value in both cases.